multinode_integration_tests/tests/verify_bill_payment.rs

```diff
@@ -24,20 +31,672 @@ use node_lib::database::db_initializer::{
+    let gas_price = 50_000_000_000_u64;
+    let gas_limit = 1_000_000_u64;
+    let tx = TransactionParameters {
+        nonce: Some(ethereum_types::U256::try_from(0).expect("Internal error")),
+        to: None,
+        gas: ethereum_types::U256::try_from(gas_limit).expect("Internal error"),
+        gas_price: Some(ethereum_types::U256::try_from(gas_price).expect("Internal error")),
+        value: ethereum_types::U256::zero(),
+        data: Bytes(data),
+        chain_id: Some(chain.rec().num_chain_id),
+    };
+    let signed_tx = sign_transaction(web3, tx, wallet);
+    match web3
+        .eth()
+        .send_raw_transaction(signed_tx.raw_transaction)
+        .wait()
+    {
+        Ok(tx_hash) => match web3.eth().transaction_receipt(tx_hash).wait() {
+            Ok(Some(tx_receipt)) => Address {
+                0: tx_receipt.contract_address.unwrap().0,
+            },
+            Ok(None) => panic!("Contract deployment failed Ok(None)"),
+            Err(e) => panic!("Contract deployment failed {:?}", e),
+        },
+        Err(e) => panic!("Contract deployment failed {:?}", e),
+    }
+}
+
+fn transfer_service_fee_amount_to_address(
+    contract_addr: Address,
+    from_wallet: &Wallet,
+    to_wallet: &Wallet,
+    amount_minor: u128,
+    transaction_nonce: u64,
+    web3: &Web3<Http>,
+    chain: Chain,
+) {
+    let data = transaction_data_web3(to_wallet, amount_minor);
+    let gas_price = 150_000_000_000_u64;
+    let gas_limit = 1_000_000_u64;
+    let tx = TransactionParameters {
+        nonce: Some(ethereum_types::U256::try_from(transaction_nonce).expect("Internal error")),
+        to: Some(contract_addr),
+        gas: ethereum_types::U256::try_from(gas_limit).expect("Internal error"),
+        gas_price: Some(ethereum_types::U256::try_from(gas_price).expect("Internal error")),
+        value: ethereum_types::U256::zero(),
+        data: Bytes(data.to_vec()),
+        chain_id: Some(chain.rec().num_chain_id),
+    };
+
+    let signed_tx = sign_transaction(web3, tx, from_wallet);
+
+    match web3
+        .eth()
+        .send_raw_transaction(signed_tx.raw_transaction)
+        .wait()
+    {
+        Ok(tx_hash) => eprintln!(
+            "Transaction {:?} of {} wei of MASQ was sent from wallet {} to {}",
+            tx_hash, amount_minor, from_wallet, to_wallet
+        ),
+        Err(e) => panic!("Transaction for token transfer failed {:?}", e),
+    }
+}
+
+fn sign_transaction(
+    web3: &Web3<Http>,
+    tx: TransactionParameters,
+    signing_wallet: &Wallet,
+) -> SignedTransaction {
+    web3.accounts()
+        .sign_transaction(
+            tx,
+            &signing_wallet
+                .prepare_secp256k1_secret()
+                .expect("wallet without secret"),
+        )
+        .wait()
+        .expect("transaction preparation failed")
+}
+
+fn transfer_transaction_fee_amount_to_address(
```

Comment:
I think you can merge the functions `transfer_transaction_fee_amount_to_address` and `transfer_service_fee_amount_to_address` inside a single function called `transfer_amount`.

It should take the `from` and `to` in the form of `Address` instead of being a wallet. That way you can either pick a wallet address or contract address.

I also checked regarding the `unlock()` method. It's weird that you need in one and not the other even though you need to transfer money in both of these functions. Upon looking further, it's safe to use `unlock()` for both cases since this is not the production code, so I think you can merge them accordingly.